cmd/kube-apiserver/app/options/options.go

```diff
@@ -231,7 +231,8 @@ func (s *ServerRunOptions) Flags() (fss cliflag.NamedFlagSets) {
 
 	fs.StringVar(&s.ServiceClusterIPRanges, "service-cluster-ip-range", s.ServiceClusterIPRanges, ""+
 		"A CIDR notation IP range from which to assign service cluster IPs. This must not "+
-		"overlap with any IP ranges assigned to nodes or pods. Max of two dual-stack CIDRs is allowed.")
+		"overlap with any IP ranges assigned to nodes or pods. Max of two dual-stack CIDRs is allowed. "+
+		"The largest supported service subnet is /12 for IPv4 and /108 for IPv6.")
```

Comment:
puff, this has changed so many times, this was /112 for IPv6, there was this in-congruence that  for IPv4 the bitmap can go to 2^20 but for IPv6 it's capped at 2^16 ...
https://github.com/kubernetes/kubernetes/blob/e640a01219736dd051dc3a9b8ac4630ec18b6b25/pkg/registry/core/service/ipallocator/allocator.go#L83-L94

However, it seems that has been corrected in validation, and allows masks until 20 bits ... but the allocator for IPv6 is still capped at 16 bits :thinking: 

@danwinship @thockin it seems that we should change the allocator maximum to 2^20 = 1048576 to make it real that the limit is /108

Reply:
🤦‍♂️

The mask must be >= 108, but the largest supported subnet is /112.